src/kernel/include/pgtbl.h

```diff
@@ -40,12 +28,58 @@ typedef enum {
 	PGTBL_INTERN_DEF = PGTBL_USER_DEF,
 } pgtbl_flags_t;
 
-#define PGTBL_PAGEIDX_SHIFT (12)
-#define PGTBL_FRAME_BITS (32 - PGTBL_PAGEIDX_SHIFT)
-#define PGTBL_FLAG_MASK ((1 << PGTBL_PAGEIDX_SHIFT) - 1)
-#define PGTBL_FRAME_MASK (~PGTBL_FLAG_MASK)
-#define PGTBL_DEPTH 2
-#define PGTBL_ORD 10
+#include "shared/cos_errno.h"
+#include "ertrie.h"
+#include "shared/util.h"
+#include "captbl.h"
+#include "retype_tbl.h"
+#include "liveness_tbl.h"
+#include "chal/defs.h"
+#include "chal/chal_prot.h"
+
+#ifndef LINUX_TEST
+#include "chal.h"
+#endif
+
+/* The page size definitions for composite OS */
+typedef enum {
+	PGSZ_128B = 7,
+	PGSZ_256B = 8,
+	PGSZ_512B = 9,
+	PGSZ_1K	  = 10,
+	PGSZ_2K	  = 11,
+	PGSZ_4K	  = 12,
+	PGSZ_8K	  = 13,
+	PGSZ_16K  = 14,
+	PGSZ_32K  = 15,
+	PGSZ_64K  = 16,
+	PGSZ_128K = 17,
+	PGSZ_256K = 18,
+	PGSZ_512K = 19,
+	PGSZ_1M	  = 20,
+	PGSZ_2M	  = 21,
+	PGSZ_4M	  = 22
+} pgtbl_sz_t;
```

Comment:
Unless you expect the numbers to be out of incremental order, it is not required to spell out their assignment for the 2nd enum onwards.

Reply:
I've read through the entire patch, and I still don't understand why we need these enums.

Reply:
These enums are needed for future patches with regards to the size of the pages